modepy/modes.py

```diff
@@ -766,6 +791,28 @@ def gradients(self) -> tuple[BasisGradientType, ...]:
         Each function returns a tuple of derivatives, one per reference axis.
         """
 
+    def functions_dr(self, axis: int) -> tuple[BasisFunctionType, ...]:
```

Comment:
Still unsure about the name. Suggestions?

Reply:
Maybe just `derivatives`? `deriv_functions`?

Reply:
I agree, that's probably the sanest. Renamed to that in the latest.

Reply:
`axis_derivative_functions`? 

I'm also leaning towards `derivatives`, so that's perfectly fine by me!

Reply:
`axis_derivatives` is another option, but it's unlikely that we'd have any other type (directional?) derivative, so :shrug:. I say let's leave it as `derivatives`.